From: Felix Fietkau <nbd@nbd.name>
Subject: [PATCH] bridge: always accept EAP

Allow EAP frames to pass through bridges even in learning state. Fixes
issues with WDS.
---
Index: linux-3.18.140/net/bridge/br_input.c
===================================================================
--- linux-3.18.140.orig/net/bridge/br_input.c
+++ linux-3.18.140/net/bridge/br_input.c
@@ -84,7 +84,7 @@ int br_handle_frame_finish(struct sk_buf
 	    br_multicast_rcv(br, p, skb, vid))
 		goto drop;
 
-	if (p->state == BR_STATE_LEARNING)
+	if ((p->state == BR_STATE_LEARNING) && skb->protocol != htons(ETH_P_PAE))
 		goto drop;
 
 	BR_INPUT_SKB_CB(skb)->brdev = br->dev;
